app/features/receive/use-receive-cashu-token.tsx

```diff
@@ -0,0 +1,349 @@
+      return pubkey === null || pubkey === cashuPubKey;
+    } catch {
+      console.warn('Failed to get pubkey from proof', proof);
+      return false;
+    }
+  });
+
+  return { claimableProofs, cannotClaimReason };
+};
+
+const getSelectableAccounts = (
+  sourceAccount: CashuAccount,
+  disableCrossMintSwap: boolean,
+  accounts: CashuAccount[],
+) => {
+  return disableCrossMintSwap
+    ? [sourceAccount]
+    : [
+        sourceAccount,
+        ...accounts.filter(
+          (account) =>
+            !account.isTestMint && account.mintUrl !== sourceAccount.mintUrl,
+        ),
+      ];
+};
+
+/**
+ * Get the default account to receive the token.
+ * If the source account is a test mint, then the source account will be returned.
+ * Otherwise it will return the user's default account if it is a cashu account.
+ * Otherwise it will return the source account.
+ * @param selectableAccounts - Accounts available for receiving
+ * @param sourceAccount - Account the token is from
+ * @param disableCrossMintSwap - Whether we can swap to other mints
+ * @param defaultAccount - The user's default account
+ * @returns The default account to receive the token
+ */
+const getDefaultReceiveAccount = (
+  selectableAccounts: CashuAccount[],
+  sourceAccount: CashuAccount,
+  disableCrossMintSwap: boolean,
+  defaultAccount: Account,
+): CashuAccount => {
+  return (
+    selectableAccounts.find(
+      (a) =>
+        a.mintUrl ===
+        (disableCrossMintSwap
+          ? sourceAccount.mintUrl
+          : defaultAccount.type === 'cashu' && defaultAccount.mintUrl),
+    ) ?? sourceAccount
+  );
+};
+
+const getBadges = (
+  account: CashuAccount,
+  allAccounts: CashuAccount[],
+  sourceAccount: CashuAccount,
+  defaultAccount: Account,
+) => {
+  const badges: string[] = [];
+
+  if (sourceAccount.isTestMint) {
+    badges.push('Test Mint');
+  }
+  if (account.mintUrl === sourceAccount.mintUrl) {
+    badges.push('Source');
+  }
+  if (
+    defaultAccount.type === 'cashu' &&
+    account.mintUrl === defaultAccount.mintUrl
+  ) {
+    badges.push('Default');
+  }
+  if (!allAccounts.some((a) => a.mintUrl === account.mintUrl)) {
+    badges.push('Unknown');
+  }
+
+  return badges;
+};
+/**
+ * Hook to receive a cashu token.
+ * @param token - The token to receive
+ * @param cashuPubKey - The user's cashu public key
+ * @returns Token data fetched by the query function and functions to claim the token.
+ * Data is undefined while isLoading is true.
+ */
+export function useReceiveCashuToken({
+  token,
+  cashuPubKey,
+}: UseReceiveCashuTokenProps): UseReceiveCashuTokenReturn {
+  const [isClaiming, setIsClaiming] = useState(false);
+  const { toast } = useToast();
+  const navigate = useNavigateWithViewTransition();
+  const [selectedReceiveAccount, setSelectedReceiveAccount] = useState<
+    CashuAccount | undefined
+  >();
+  const { data: allAccounts } = useAccounts();
+  const accounts = allAccounts?.filter(
```

Comment:
you can write this like this:
```
const accounts = allAccounts?.filter(
  (a): a is CashuAccount => a.type === 'cashu',
);
```

this way you tell the compiler that only CashuAccount is left after filtering and you don't have to do ugly as

Reply:
I don't think ? is needed after accounts